promql/promqltest/test.go

```diff
@@ -1651,3 +1653,107 @@ func timeMilliseconds(t time.Time) int64 {
 func durationMilliseconds(d time.Duration) int64 {
 	return int64(d / (time.Millisecond / time.Nanosecond))
 }
+
+func MigrateTestData(mode string) error {
+	const dir = "promql/promqltest/testdata"
+
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		return fmt.Errorf("failed to read testdata directory: %w", err)
+	}
+
+	strictAnnotations := map[string][]string{
+		"eval_fail":    {"  expect fail", "  expect no_warn", "  expect no_info"},
+		"eval_warn":    {"  expect no_fail", "  expect warn", "  expect no_info"},
+		"eval_info":    {"  expect info", "  expect no_warn", "  expect no_info"},
+		"eval_ordered": {"  expect ordered", "  expect no_fail", "  expect no_warn", "  expect no_info"},
+		"eval":         {"  expect no_fail", "  expect no_warn", "  expect no_info"},
+	}
+
+	basicAnnotations := map[string][]string{
+		"eval_fail":    {"  expect fail"},
+		"eval_warn":    {"  expect warn"},
+		"eval_info":    {"  expect info"},
```

Comment:
To be really relaxed, these two should actually be empty (the new `eval` tolerates annotations).

Maybe we should have three modes: `tolerant` (converting `eval_warn` and `eval_info` to a simple `eval`), `basic` (as implemented here), and `strict` (as implemented above).